snakemake/scheduler.py

```
@@ -737,6 +737,113 @@ def calc_reward():
+            # Iterate max_sorted, picking the last (most rewarding) element at a time,
+            # until we have either picked all elements, or exhausted a resource.
+            solution = set()
+            while max_sorted:
+                # Get the next best job and its resources.
+                reward, job = max_sorted.pop()
+                job_res = self.job_weight(job)
+                assert len(job_res) == len(max_glob_res)
+
+                # Check resources
+                exhausted_some_res = any(
+                    used_res[i] + job_res[i] >= res[i] for i in range(len(res))
```

Ruff

Undefined name `res`

(F821)